### PR TITLE
added image-uploader component and refactored profile settings page

### PR DESCRIPTION
**Pull request by alishaz-polymath**
Pull Request Details:

1. Adjusted Avatar.tsx component to allow state management in settings/profile page to re-render the avatar upon avatar upload using the ImageUploader.tsx component
avatar

2. Added ImageUploader.tsx component (which uses react-easy-crop plugin) for uploading images (avatar, logo, etc.)
upload

3. Minor improvement to the add-guest button in booking confirmation page
4. Added Image Uploader to the settings/profile page and hooked it up with avatar upload flow
5. Updated the settings/profile page to match better with the latest design reference
image

### CODE REVIEW

**Comment by mihaic195** [comment]
Where is the image uploaded?

**Comment by PeerRich** [comment]
> Where is the image uploaded?

afaik it's saved as base64 in your local db since the avatar is really small (100x100)

File: components/Avatar.tsx [hunk 1]
```diff
@@ -1,13 +1,18 @@
...
     fallback?: JSX.Element;
+    imageSrc?: string;
 }) {
     const [gravatarAvailable, setGravatarAvailable] = useState(true);
 
+    if (imageSrc) {
+        return <img src={imageSrc} alt="Avatar" className={className} />;
+    }
+
     if (user.avatar) {
```
Comment (emrysal) [inline components/Avatar.tsx:16]: Do you still need this?

File: components/ImageUploader.tsx [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+      <button
+      type="button"
+      className="ml-4 cursor-pointer inline-flex items-center px-4 py-1 border border-gray-300 shadow-sm text-xs font-medium rounded-sm text-gray-700 bg-white hover:bg-gray-50 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-neutral-500;"
+      onClick={openUploaderModal}
+      >
+        {buttonMsg}
+      </button>
+
+      {
+        imageUploadModalOpen &&
```
Comment (emrysal) [inline components/ImageUploader.tsx:144]: Good for now, but have a look at components/Dialog.tsx

File: components/ImageUploader.tsx [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+    imageRef ? (setIsImageShown(true), setShownImage(imageRef)) : setIsImageShown(false);
+    setImageUploadModalOpen(!imageUploadModalOpen)
+  }
+  
+  const closeImageUploadModal = () => {
+    setImageUploadModalOpen(false);
+  };
+
+  async function ImageUploadHandler() {
+    console.log(imageFileRef.current.files[0]);
```
Comment (emrysal) [inline components/ImageUploader.tsx:28]: Some debug remnants to remove

**Review by emrysal — APPROVED** [review]
This is very good! couple of questions, please remove the console.log remnants - other than that, all good to go!

image

Great!

**Review by PeerRich — APPROVED** [review]
great PR!!

**Review by PeerRich — APPROVED** [review]

**Comment by alishaz-polymath** [comment]
> 
> 
> great PR!!

Thanks @PeerRich !

**Pull request merged**